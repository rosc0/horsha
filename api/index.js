import config from 'react-native-config'
import { build, serialize, emptyToNull } from '@application/utils'
import store from '@application/store'
import TimerMixin from 'react-timer-mixin'
import { setAlertStatus } from '../actions/app'
// import { logoutCurrentUser } from '../actions'
import * as k from '@constants/storageKeys'
import { AsyncStorage } from 'react-native'
// import Authentication from '@api/auth'

const title = 'There’s been a server error'
const subTitle = 'Something’s wrong, please try again later'

// const Auth = new Authentication()

class API {
  constructor(props) {
    this._config = new WeakMap()
    this._endpoint = new WeakMap()
    this._hasBaseURL = false

    if (!props || !props.isAuth) this._setup()
  }

  _setup() {
    const headers = {
      'Content-Type': 'application/json',
    }

    this._config.set(this, { headers })
  }

  setAuthConfig() {
    const headers = {
      'Content-Type': 'application/x-www-form-urlencoded',
    }

    this._config.set(this, { headers })
    return this
  }

  setContentType(contentType) {
    if (contentType) {
      const headers = {
        'Content-Type': contentType,
      }

      this._config.set(this, { headers })
    }
    return this
  }

  _setMethod(method) {
    const config = this._config.get(this) || {}
    config.method = method

    this._config.set(this, config)
  }

  setEndpoint(endpoint, hasBaseURL = false) {
    this._endpoint.set(this, endpoint)

    this._hasBaseURL = hasBaseURL
    return this
  }

  getEndpointPath() {
    return this._endpoint.get(this)
  }

  getEndpoint() {
    if (this._hasBaseURL) {
      return this._endpoint.get(this)
    }

    return config.API_BASE + this._endpoint.get(this)
  }

  getHeaders = () => {
    const state = store.getState()

    const {
      auth: { auth },
      user: { user },
    } = state

    return {
      access_token: auth.accessToken,
      user_id: user.id,
    }
  }

  setBody(body, isJSON) {
    const config = this._config.get(this) || {}

    config.body = isJSON
      ? JSON.stringify(emptyToNull(body))
      : serialize(emptyToNull(body))

    this._config.set(this, config)

    return this
  }

  getConfig() {
    let _config = this._config.get(this)

    const isAuth =
      _config.headers['Content-Type'] === 'application/x-www-form-urlencoded'

    if (isAuth) {
      let isClientIdDuplicated = false

      if (_config.body && !_config.body._parts) {
        isClientIdDuplicated = _config.body.indexOf('client_id') > -1
      }

      if (
        _config.method !== 'GET' &&
        isAuth &&
        !isClientIdDuplicated &&
        !_config.body._parts
      ) {
        _config.body += '&client_id=' + config.CLIENT_ID
      }
    }

    return this._config.get(this)
  }

  _makeRequest() {
    return new Promise((resolve, reject) => {
      let config = this.getConfig()
      let endpoint = this.getEndpoint()

      if (config.method === 'GET') {
        delete config.body
      }

      fetch(endpoint, config)
        .then(response => {
          TimerMixin.setTimeout(() => null, 0)

          if (response.status === 500) {
            // to handler if API return is not valid
            console.warn('throw Error', response)
            store.dispatch(setAlertStatus('error', title, subTitle))
            throw Error(response.statusText)
          }

          if (response.status === 403) {
            console.warn('logged out', response)
            if (
              response._bodyInit !==
              '{"status":403,"description":"User not allowed to allowed to see pictures for horse"}'
            ) {
              AsyncStorage.removeItem(k.AUTH_NEW_TOKENS)

              return store.dispatch({
                type: 'RESET',
                payload: {
                  app: store.getState().app,
                },
              })
            }
            // return response
            AsyncStorage.removeItem(k.AUTH_NEW_TOKENS)

            return store.dispatch({
              type: 'RESET',
              payload: {
                app: store.getState().app,
              },
            })
          }

          if (response.status === 401) {
            if (
              response._bodyInit === '{"code":401,"error":"Invalid token"} '
            ) {
              // ugly workaround
              AsyncStorage.removeItem(k.AUTH_NEW_TOKENS)
              return store.dispatch({
                type: 'RESET',
                payload: {
                  app: store.getState().app,
                },
              })
            }

            return response
          }

          if (response.status === 422) {
            if (
              response._bodyInit === '{"code":422,"text":"Missing user_id"}' ||
              response._bodyInit ===
                '{"status":422,"description":"Invalid UUID string: null"}'
            ) {
              // ugly workaround
              AsyncStorage.removeItem(k.AUTH_NEW_TOKENS)
              return store.dispatch({
                type: 'RESET',
                payload: {
                  app: store.getState().app,
                },
              })
            }

            return response
          }

          if (response.status < 200 || response.status > 299) {
            // to handler if API return is not valid
            console.warn('throw Error', response)
            store.dispatch(setAlertStatus('error', title, subTitle))
            throw Error('Error')
          }

          if (response.status === 204) {
            return response
          }

          const isResponseJSON =
            response.headers.get('content-type') &&
            response.headers.get('content-type').indexOf('application/json') !==
              -1

          return (response.status === 200 || response.status === 201) &&
            isResponseJSON
            ? response.json()
            : response
        })
        .then(res => {
          resolve(res)
        })
        .catch(err => {
          reject(err)
        })

      this._setup()
    })
  }

  addToken(type) {
    const { auth } = store.getState().auth
    const { accessToken: token, expireDate: expires } = auth

    const method =
      type === 'qs'
        ? this.addQueryStringToken.bind(this)
        : this.addJSONToken.bind(this)

    if (token && expires > +new Date()) {
      method(token)
    }
  }

  addQueryStringToken(token) {
    const body = this.getEndpointPath()
    const endpoint = build('get', { body, token })

    this.setEndpoint(endpoint)
  }

  addJSONToken(token) {
    const { body } = this.getConfig()
    const data = build('post', { body, token })

    this._config.set(this, { body: data })
  }

  get(hasToken) {
    if (hasToken) this.addToken('qs')

    this._setMethod('GET')
    this.setBody(null)
    return this._makeRequest()
  }

  post(hasToken) {
    if (hasToken) this.addToken('json')

    this._setMethod('POST')
    return this._makeRequest()
  }

  put(hasToken) {
    if (hasToken) this.addToken('json')

    this._setMethod('PUT')
    return this._makeRequest()
  }

  delete(hasToken, type = 'qs') {
    if (hasToken) this.addToken(type)

    this._setMethod('DELETE')
    return this._makeRequest()
  }
}

export default API
