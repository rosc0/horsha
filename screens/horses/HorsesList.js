import React, { PureComponent } from 'react'
import { FlatList, StyleSheet, View } from 'react-native'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import Horse from './components/Horse'
import Menu from './components/Menu'
import EmptyHorseList from './components/EmptyHorseList'
import AddButton from '@components/AddButton'
import HeaderTitle from '@components/HeaderTitle'
import NotificationsHeader from '@components/NotificationsHeader'
import { theme } from '@styles/theme'

import * as horsesActions from '@actions/horses'

class HorsesList extends PureComponent {
  state = {
    selectedHorse: null,
  }

  static navigationOptions = ({ navigation }) => {
    return {
      headerTitle: <HeaderTitle title={'horses/horses'} />,
      headerRight: (
        <View style={styles.header}>
          <AddButton
            imageStyle={styles.addButton}
            onPress={() => navigation.navigate('AddHorse')}
          />
          <NotificationsHeader fromRoute="Horses" navigation={navigation} />
        </View>
      ),
    }
  }

  static getDerivedStateFromProps(props, state) {
    if (props.horses.horse && !state.selectedHorse) {
      return {
        selectedHorse: props.horses.horse,
      }
    }

    // Return null to indicate no change to state.
    return null
  }

  componentDidMount() {
    this.props.actions.getHorses()
  }

  getMoreHorses = () => {
    const { cursor, horsesRemaining } = this.props.horses

    if (horsesRemaining !== 0) {
      this.props.actions.getMoreHorses(cursor)
    }
  }

  handleGoToRoute = (route, params) => () =>
    this.props.navigation.navigate(route, params)

  handleSelectHorse = async horseUser => {
    if (horseUser.horse === this.state.selectedHorse) return

    if (this.props.horses.horses.length) {
      this.horseListRef.scrollToOffset({ x: 0, animated: true })
    }

    this.props.actions.setLastAccessedHorse(horseUser.id)

    this.setState({ selectedHorse: horseUser.horse })
  }

  render() {
    const { selectedHorse } = this.state
    const { horses } = this.props.horses

    return (
      <View style={styles.container}>
        {this.props.horses.horses.length === 0 ? (
          <EmptyHorseList />
        ) : (
          <React.Fragment>
            <View>
              <FlatList
                ref={ref => (this.horseListRef = ref)}
                style={styles.horsesContainer}
                data={horses}
                horizontal={true}
                keyExtractor={horse => horse.id}
                onEndReachedThreshold={2}
                onEndReached={this.getMoreHorses}
                showsHorizontalScrollIndicator={false}
                renderItem={({ item, index }) => (
                  <Horse
                    horse={item.horse}
                    archived={item.relation_type === 'archived'}
                    selectedHorse={index === 0}
                    onSelectHorse={() => this.handleSelectHorse(item)}
                  />
                )}
              />
            </View>

            {selectedHorse && (
              <View style={styles.selectedHorseContainer}>
                <Menu
                  horse={this.props.horses.horse}
                  goToRoute={this.handleGoToRoute}
                />
              </View>
            )}
          </React.Fragment>
        )}
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: theme.backgroundColor,
    flex: 1,
  },
  horsesContainer: {
    paddingTop: 5,
    paddingHorizontal: 10,
    backgroundColor: 'white',
  },
  selectedHorseContainer: {
    flex: 1,
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
  addButton: {
    marginRight: 10,
  },
})

export default connect(
  ({ horses }) => ({
    horses,
  }),
  dispatch => ({
    actions: bindActionCreators(horsesActions, dispatch),
  })
)(HorsesList)
