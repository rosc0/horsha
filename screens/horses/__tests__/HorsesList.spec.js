import React from 'react'
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'
import promiseMiddleware from 'redux-promise-middleware'
import thunk from 'redux-thunk'

import horsesReducers, { initialState as horses } from '@reducers/horses'

import { horse_user } from '@test/api'
import types from '@constants/actionTypes'
import HorsesList from '../HorsesList'

import { current_user } from '../../../../test/api'

const mockStore = configureStore([thunk, promiseMiddleware()])
const state = { horses: {}, user: current_user.GET }
let store = mockStore(state)
console.log('state', state)
beforeEach(() => {
  // Resets store before each test
  store = mockStore(state)
})

const setupComponent = () => shallow(<HorsesList />, { context: { store } })

describe('<HorsesList />', () => {
  it('renders loading', () => {
    const component = setupComponent()
    expect(component.dive()).toMatchSnapshot()
  })

  it('renders without exploding', async () => {
    const horses = horsesReducers(state.horses, {
      type: types.GET_HORSES_FULFILLED,
      payload: horse_user.GET,
    })

    store = mockStore({
      horses: {
        ...horses,
        horse: horses[0],
      },
    })

    const component = setupComponent()
    expect(component.dive()).toMatchSnapshot()
  })
})
