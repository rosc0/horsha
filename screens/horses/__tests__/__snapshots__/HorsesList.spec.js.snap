// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HorsesList /> renders loading 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HorsesList
    actions={
      Object {
        "addFriendToTeam": [Function],
        "addHorse": [Function],
        "addHorseImage": [Function],
        "clearAddHorse": [Function],
        "clearSelectedImages": [Function],
        "clearSelectedRides": [Function],
        "deleteHorseImages": [Function],
        "deleteHorseJournal": [Function],
        "editHorseImage": [Function],
        "editHorseJournal": [Function],
        "editHorseProfile": [Function],
        "editHorseUser": [Function],
        "editShareScope": [Function],
        "getHorseAlbum": [Function],
        "getHorseJournal": [Function],
        "getHorseProfile": [Function],
        "getHorseUser": [Function],
        "getHorses": [Function],
        "getJournalPost": [Function],
        "getMoreHorses": [Function],
        "getStatusPost": [Function],
        "saveHorseJournal": [Function],
        "selectRide": [Function],
        "setHorse": [Function],
        "setLastAccessedHorse": [Function],
        "setNextStep": [Function],
        "setPreviousStep": [Function],
        "toggleSelectImage": [Function],
        "unselectRide": [Function],
      }
    }
    horses={
      Object {
        "addHorse": Object {
          "fetching": false,
          "step": "NAME",
        },
        "album": Object {
          "fetching": true,
          "pictures": Array [],
          "selectedImages": Array [],
          "upload": Object {
            "errored": false,
            "isUploading": false,
            "uploaded": false,
          },
        },
        "cursor": null,
        "detail": Object {
          "entry": null,
          "error": false,
          "fetched": false,
          "fetching": false,
        },
        "error": false,
        "fetched": false,
        "fetching": false,
        "forceUpdate": null,
        "horse": Object {
          "id": null,
          "image": null,
          "name": null,
          "relationType": null,
        },
        "horsePictureChanged": 0,
        "horses": Array [],
        "horsesRemaining": 0,
        "journal": Object {
          "error": false,
          "fetched": false,
          "fetching": false,
        },
        "profile": Object {
          "birth": null,
          "breed": null,
          "color": null,
          "description": null,
          "fetched": false,
          "fetching": true,
          "gender": null,
          "height": null,
          "horseCare": null,
          "image": "",
          "name": null,
          "stats": Object {
            "distance": 0,
            "duration": 0,
            "rides": 0,
          },
          "weight": null,
        },
        "selectedRides": Array [],
        "team": Object {
          "fetching": true,
          "owners": Array [],
          "users": Array [],
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Loading
        type="spinner"
      />,
      "style": Object {
        "backgroundColor": "white",
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "type": "spinner",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Loading
          type="spinner"
        />,
        "style": Object {
          "backgroundColor": "white",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "type": "spinner",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;

exports[`<HorsesList /> renders without exploding 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HorsesList
    actions={
      Object {
        "addFriendToTeam": [Function],
        "addHorse": [Function],
        "addHorseImage": [Function],
        "clearAddHorse": [Function],
        "clearSelectedImages": [Function],
        "clearSelectedRides": [Function],
        "deleteHorseImages": [Function],
        "deleteHorseJournal": [Function],
        "editHorseImage": [Function],
        "editHorseJournal": [Function],
        "editHorseProfile": [Function],
        "editHorseUser": [Function],
        "editShareScope": [Function],
        "getHorseAlbum": [Function],
        "getHorseJournal": [Function],
        "getHorseProfile": [Function],
        "getHorseUser": [Function],
        "getHorses": [Function],
        "getJournalPost": [Function],
        "getMoreHorses": [Function],
        "getStatusPost": [Function],
        "saveHorseJournal": [Function],
        "selectRide": [Function],
        "setHorse": [Function],
        "setLastAccessedHorse": [Function],
        "setNextStep": [Function],
        "setPreviousStep": [Function],
        "toggleSelectImage": [Function],
        "unselectRide": [Function],
      }
    }
    horses={
      Object {
        "addHorse": Object {
          "fetching": false,
          "step": "NAME",
        },
        "album": Object {
          "fetching": true,
          "pictures": Array [],
          "selectedImages": Array [],
          "upload": Object {
            "errored": false,
            "isUploading": false,
            "uploaded": false,
          },
        },
        "cursor": undefined,
        "detail": Object {
          "entry": null,
          "error": false,
          "fetched": false,
          "fetching": false,
        },
        "error": false,
        "fetched": true,
        "fetching": false,
        "forceUpdate": null,
        "horse": undefined,
        "horsePictureChanged": 0,
        "horses": Array [
          Object {
            "date_created": 1465208196000,
            "date_last_accessed": 1501745884000,
            "horse": Object {
              "date_created": 1464792727000,
              "id": "8f13609e-9872-4f90-8deb-3c4dc0a9bd02",
              "profile": Object {
                "name": "wicked long name test hello",
              },
              "profile_picture": Object {
                "fullsize": Object {
                  "height": 800,
                  "url": "https://dev-content.horsha.com/023f45bb-c202-47cd-8ada-45f66c9167fd.jpg",
                  "width": 800,
                },
                "thumbnail": Object {
                  "height": 300,
                  "url": "https://dev-content.horsha.com/3172cfe9-4b5b-48a4-8dd3-5a10f58c61fc.jpg",
                  "width": 300,
                },
              },
            },
            "id": "e97d2af9-1365-44a9-873d-2b55198ea815",
            "relation_type": "sharer",
            "user": Object {
              "id": "95abaa8c-98f6-4eb5-9c84-a7b4abd19d4b",
              "name": "Do",
              "profile_picture": Object {
                "fullsize": Object {
                  "height": 1286,
                  "url": "https://dev-content.horsha.com/f60b0bb9-7ace-4ee4-959d-4a2a8a219d55.jpg",
                  "width": 1600,
                },
                "thumbnail": Object {
                  "height": 300,
                  "url": "https://dev-content.horsha.com/70d44fd6-6a4f-42d7-a912-04354045defb.jpg",
                  "width": 300,
                },
              },
            },
          },
        ],
        "horsesRemaining": 45,
        "journal": Object {
          "error": false,
          "fetched": false,
          "fetching": false,
        },
        "profile": Object {
          "birth": null,
          "breed": null,
          "color": null,
          "description": null,
          "fetched": false,
          "fetching": true,
          "gender": null,
          "height": null,
          "horseCare": null,
          "image": "",
          "name": null,
          "stats": Object {
            "distance": 0,
            "duration": 0,
            "rides": 0,
          },
          "weight": null,
        },
        "selectedRides": Array [],
        "team": Object {
          "fetching": true,
          "owners": Array [],
          "users": Array [],
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Component>
          <FlatList
            data={
              Array [
                Object {
                  "date_created": 1465208196000,
                  "date_last_accessed": 1501745884000,
                  "horse": Object {
                    "date_created": 1464792727000,
                    "id": "8f13609e-9872-4f90-8deb-3c4dc0a9bd02",
                    "profile": Object {
                      "name": "wicked long name test hello",
                    },
                    "profile_picture": Object {
                      "fullsize": Object {
                        "height": 800,
                        "url": "https://dev-content.horsha.com/023f45bb-c202-47cd-8ada-45f66c9167fd.jpg",
                        "width": 800,
                      },
                      "thumbnail": Object {
                        "height": 300,
                        "url": "https://dev-content.horsha.com/3172cfe9-4b5b-48a4-8dd3-5a10f58c61fc.jpg",
                        "width": 300,
                      },
                    },
                  },
                  "id": "e97d2af9-1365-44a9-873d-2b55198ea815",
                  "relation_type": "sharer",
                  "user": Object {
                    "id": "95abaa8c-98f6-4eb5-9c84-a7b4abd19d4b",
                    "name": "Do",
                    "profile_picture": Object {
                      "fullsize": Object {
                        "height": 1286,
                        "url": "https://dev-content.horsha.com/f60b0bb9-7ace-4ee4-959d-4a2a8a219d55.jpg",
                        "width": 1600,
                      },
                      "thumbnail": Object {
                        "height": 300,
                        "url": "https://dev-content.horsha.com/70d44fd6-6a4f-42d7-a912-04354045defb.jpg",
                        "width": 300,
                      },
                    },
                  },
                },
              ]
            }
            disableVirtualization={false}
            horizontal={true}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onEndReached={[Function]}
            onEndReachedThreshold={2}
            renderItem={[Function]}
            scrollEventThrottle={50}
            style={
              Object {
                "marginLeft": 15,
                "marginTop": 5,
              }
            }
            updateCellsBatchingPeriod={50}
            windowSize={21}
          />
        </Component>,
        null,
      ],
      "style": Object {
        "backgroundColor": "white",
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <FlatList
            data={
              Array [
                Object {
                  "date_created": 1465208196000,
                  "date_last_accessed": 1501745884000,
                  "horse": Object {
                    "date_created": 1464792727000,
                    "id": "8f13609e-9872-4f90-8deb-3c4dc0a9bd02",
                    "profile": Object {
                      "name": "wicked long name test hello",
                    },
                    "profile_picture": Object {
                      "fullsize": Object {
                        "height": 800,
                        "url": "https://dev-content.horsha.com/023f45bb-c202-47cd-8ada-45f66c9167fd.jpg",
                        "width": 800,
                      },
                      "thumbnail": Object {
                        "height": 300,
                        "url": "https://dev-content.horsha.com/3172cfe9-4b5b-48a4-8dd3-5a10f58c61fc.jpg",
                        "width": 300,
                      },
                    },
                  },
                  "id": "e97d2af9-1365-44a9-873d-2b55198ea815",
                  "relation_type": "sharer",
                  "user": Object {
                    "id": "95abaa8c-98f6-4eb5-9c84-a7b4abd19d4b",
                    "name": "Do",
                    "profile_picture": Object {
                      "fullsize": Object {
                        "height": 1286,
                        "url": "https://dev-content.horsha.com/f60b0bb9-7ace-4ee4-959d-4a2a8a219d55.jpg",
                        "width": 1600,
                      },
                      "thumbnail": Object {
                        "height": 300,
                        "url": "https://dev-content.horsha.com/70d44fd6-6a4f-42d7-a912-04354045defb.jpg",
                        "width": 300,
                      },
                    },
                  },
                },
              ]
            }
            disableVirtualization={false}
            horizontal={true}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onEndReached={[Function]}
            onEndReachedThreshold={2}
            renderItem={[Function]}
            scrollEventThrottle={50}
            style={
              Object {
                "marginLeft": 15,
                "marginTop": 5,
              }
            }
            updateCellsBatchingPeriod={50}
            windowSize={21}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Array [
              Object {
                "date_created": 1465208196000,
                "date_last_accessed": 1501745884000,
                "horse": Object {
                  "date_created": 1464792727000,
                  "id": "8f13609e-9872-4f90-8deb-3c4dc0a9bd02",
                  "profile": Object {
                    "name": "wicked long name test hello",
                  },
                  "profile_picture": Object {
                    "fullsize": Object {
                      "height": 800,
                      "url": "https://dev-content.horsha.com/023f45bb-c202-47cd-8ada-45f66c9167fd.jpg",
                      "width": 800,
                    },
                    "thumbnail": Object {
                      "height": 300,
                      "url": "https://dev-content.horsha.com/3172cfe9-4b5b-48a4-8dd3-5a10f58c61fc.jpg",
                      "width": 300,
                    },
                  },
                },
                "id": "e97d2af9-1365-44a9-873d-2b55198ea815",
                "relation_type": "sharer",
                "user": Object {
                  "id": "95abaa8c-98f6-4eb5-9c84-a7b4abd19d4b",
                  "name": "Do",
                  "profile_picture": Object {
                    "fullsize": Object {
                      "height": 1286,
                      "url": "https://dev-content.horsha.com/f60b0bb9-7ace-4ee4-959d-4a2a8a219d55.jpg",
                      "width": 1600,
                    },
                    "thumbnail": Object {
                      "height": 300,
                      "url": "https://dev-content.horsha.com/70d44fd6-6a4f-42d7-a912-04354045defb.jpg",
                      "width": 300,
                    },
                  },
                },
              },
            ],
            "disableVirtualization": false,
            "horizontal": true,
            "initialNumToRender": 10,
            "keyExtractor": [Function],
            "maxToRenderPerBatch": 10,
            "numColumns": 1,
            "onEndReached": [Function],
            "onEndReachedThreshold": 2,
            "renderItem": [Function],
            "scrollEventThrottle": 50,
            "style": Object {
              "marginLeft": 15,
              "marginTop": 5,
            },
            "updateCellsBatchingPeriod": 50,
            "windowSize": 21,
          },
          "ref": [Function],
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      null,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Component>
            <FlatList
              data={
                Array [
                  Object {
                    "date_created": 1465208196000,
                    "date_last_accessed": 1501745884000,
                    "horse": Object {
                      "date_created": 1464792727000,
                      "id": "8f13609e-9872-4f90-8deb-3c4dc0a9bd02",
                      "profile": Object {
                        "name": "wicked long name test hello",
                      },
                      "profile_picture": Object {
                        "fullsize": Object {
                          "height": 800,
                          "url": "https://dev-content.horsha.com/023f45bb-c202-47cd-8ada-45f66c9167fd.jpg",
                          "width": 800,
                        },
                        "thumbnail": Object {
                          "height": 300,
                          "url": "https://dev-content.horsha.com/3172cfe9-4b5b-48a4-8dd3-5a10f58c61fc.jpg",
                          "width": 300,
                        },
                      },
                    },
                    "id": "e97d2af9-1365-44a9-873d-2b55198ea815",
                    "relation_type": "sharer",
                    "user": Object {
                      "id": "95abaa8c-98f6-4eb5-9c84-a7b4abd19d4b",
                      "name": "Do",
                      "profile_picture": Object {
                        "fullsize": Object {
                          "height": 1286,
                          "url": "https://dev-content.horsha.com/f60b0bb9-7ace-4ee4-959d-4a2a8a219d55.jpg",
                          "width": 1600,
                        },
                        "thumbnail": Object {
                          "height": 300,
                          "url": "https://dev-content.horsha.com/70d44fd6-6a4f-42d7-a912-04354045defb.jpg",
                          "width": 300,
                        },
                      },
                    },
                  },
                ]
              }
              disableVirtualization={false}
              horizontal={true}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              numColumns={1}
              onEndReached={[Function]}
              onEndReachedThreshold={2}
              renderItem={[Function]}
              scrollEventThrottle={50}
              style={
                Object {
                  "marginLeft": 15,
                  "marginTop": 5,
                }
              }
              updateCellsBatchingPeriod={50}
              windowSize={21}
            />
          </Component>,
          null,
        ],
        "style": Object {
          "backgroundColor": "white",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <FlatList
              data={
                Array [
                  Object {
                    "date_created": 1465208196000,
                    "date_last_accessed": 1501745884000,
                    "horse": Object {
                      "date_created": 1464792727000,
                      "id": "8f13609e-9872-4f90-8deb-3c4dc0a9bd02",
                      "profile": Object {
                        "name": "wicked long name test hello",
                      },
                      "profile_picture": Object {
                        "fullsize": Object {
                          "height": 800,
                          "url": "https://dev-content.horsha.com/023f45bb-c202-47cd-8ada-45f66c9167fd.jpg",
                          "width": 800,
                        },
                        "thumbnail": Object {
                          "height": 300,
                          "url": "https://dev-content.horsha.com/3172cfe9-4b5b-48a4-8dd3-5a10f58c61fc.jpg",
                          "width": 300,
                        },
                      },
                    },
                    "id": "e97d2af9-1365-44a9-873d-2b55198ea815",
                    "relation_type": "sharer",
                    "user": Object {
                      "id": "95abaa8c-98f6-4eb5-9c84-a7b4abd19d4b",
                      "name": "Do",
                      "profile_picture": Object {
                        "fullsize": Object {
                          "height": 1286,
                          "url": "https://dev-content.horsha.com/f60b0bb9-7ace-4ee4-959d-4a2a8a219d55.jpg",
                          "width": 1600,
                        },
                        "thumbnail": Object {
                          "height": 300,
                          "url": "https://dev-content.horsha.com/70d44fd6-6a4f-42d7-a912-04354045defb.jpg",
                          "width": 300,
                        },
                      },
                    },
                  },
                ]
              }
              disableVirtualization={false}
              horizontal={true}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              numColumns={1}
              onEndReached={[Function]}
              onEndReachedThreshold={2}
              renderItem={[Function]}
              scrollEventThrottle={50}
              style={
                Object {
                  "marginLeft": 15,
                  "marginTop": 5,
                }
              }
              updateCellsBatchingPeriod={50}
              windowSize={21}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "data": Array [
                Object {
                  "date_created": 1465208196000,
                  "date_last_accessed": 1501745884000,
                  "horse": Object {
                    "date_created": 1464792727000,
                    "id": "8f13609e-9872-4f90-8deb-3c4dc0a9bd02",
                    "profile": Object {
                      "name": "wicked long name test hello",
                    },
                    "profile_picture": Object {
                      "fullsize": Object {
                        "height": 800,
                        "url": "https://dev-content.horsha.com/023f45bb-c202-47cd-8ada-45f66c9167fd.jpg",
                        "width": 800,
                      },
                      "thumbnail": Object {
                        "height": 300,
                        "url": "https://dev-content.horsha.com/3172cfe9-4b5b-48a4-8dd3-5a10f58c61fc.jpg",
                        "width": 300,
                      },
                    },
                  },
                  "id": "e97d2af9-1365-44a9-873d-2b55198ea815",
                  "relation_type": "sharer",
                  "user": Object {
                    "id": "95abaa8c-98f6-4eb5-9c84-a7b4abd19d4b",
                    "name": "Do",
                    "profile_picture": Object {
                      "fullsize": Object {
                        "height": 1286,
                        "url": "https://dev-content.horsha.com/f60b0bb9-7ace-4ee4-959d-4a2a8a219d55.jpg",
                        "width": 1600,
                      },
                      "thumbnail": Object {
                        "height": 300,
                        "url": "https://dev-content.horsha.com/70d44fd6-6a4f-42d7-a912-04354045defb.jpg",
                        "width": 300,
                      },
                    },
                  },
                },
              ],
              "disableVirtualization": false,
              "horizontal": true,
              "initialNumToRender": 10,
              "keyExtractor": [Function],
              "maxToRenderPerBatch": 10,
              "numColumns": 1,
              "onEndReached": [Function],
              "onEndReachedThreshold": 2,
              "renderItem": [Function],
              "scrollEventThrottle": 50,
              "style": Object {
                "marginLeft": 15,
                "marginTop": 5,
              },
              "updateCellsBatchingPeriod": 50,
              "windowSize": 21,
            },
            "ref": [Function],
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        null,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
